src/Adapter/SearchEngine/AbstractSearchEngineHandler.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\SearchEngine;
+
+use PrestaShop\PrestaShop\Core\Domain\SearchEngine\Exception\SearchEngineException;
+use PrestaShop\PrestaShop\Core\Domain\SearchEngine\Exception\SearchEngineNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\SearchEngine\ValueObject\SearchEngineId;
+use PrestaShopException;
+use SearchEngine;
+
+abstract class AbstractSearchEngineHandler
+{
+    /**
+     * Gets legacy search engine.
+     *
+     * @param SearchEngineId $searchEngineId
+     *
+     * @return SearchEngine
+     *
+     * @throws SearchEngineException
+     */
+    protected function getSearchEngine(SearchEngineId $searchEngineId): SearchEngine
+    {
+        try {
+            $searchEngine = new SearchEngine($searchEngineId->getValue());
+        } catch (PrestaShopException $e) {
+            throw new SearchEngineException('Failed to create new search engine', 0, $e);
```

Comment:
```suggestion
            throw new SearchEngineException('Failed to retrieve new search engine', 0, $e);
```
We are not looking to create a SE, we want to retrieve the data from database 😉